python-code/code.py

```diff
@@ -0,0 +1,125 @@
+import requests
+import os
+import time
+import pandas as pd
+import glob
+import datetime as dt
+
+def verify_dirs_exist():
+    #This notebook requires a few directories
+    dirs = ["download", "download\csv", "download\excel"]
+    for d in dirs:
+        curpath = os.path.abspath(os.curdir) # get current working directory
+        full_path = os.path.join(curpath, d) # join cwd with proposed d
+        create_dir_if_not_exists(full_path)
+
+def create_dir_if_not_exists(full_path):
+    # expects a full path to the directory to test against or to create.
+    if not os.path.exists(full_path):
+        os.makedirs(full_path)
+        print("Created directory ", full_path)
+    else:
+        print("Directory ", full_path, " already existed")
+
+def generate_file_name_from_url(url):
+    month_year = url.split("\\")[-1].split("_")[-1].split(".")[0]
+    month = month_year[:2]
+    year = month_year[2:]
+    new_file_name = "ZIP-COUNTY-FIPS_"+year + "-" + month
+    return new_file_name
+
+def get_file_path(url, csv_file=False):
+    "Takes in the full url and returns the full file path"
+    "File names are ZIP_COUNTY_032010.xlsx"
+    curpath = os.path.abspath(os.curdir) #get current working directory
+    full_path = ''
+    if csv_file:
+        #If we are passing in a csv, change xlsx to .csv
+        csv_file_name = url.split("\\")[-1][:-5] + ".csv"
```

Comment:
I would use `replace` to swap `.xlsx` for `.csv` instead of using `[:-5]` and `+`. It makes it clearer to someone reading the code what you intend to do. `os.path.split(url)[-1].replace('.xlsx', '.csv')`